### DISCUSSION TITLE
Clicking on selected element in a SelectField empties the field even if isMulti is disabled

### DISCUSSION THREAD

**Discussion by mperrando** [buildo/bento-design-system#807] [Ideas]
This behaviour can be seen also in the demo page

https://playroom.bento.buildo.io/#?code=N4Igxg9gJgpiBcJgB0B2ACdAKLBKdAvAHzooabqSoDOALugNoBuAhgDYCuMANOtTLQBq7LgF1C6AEowWYWgDoO-AMq0WtGFg6pYAMwCWqGFFxoK6AE4COFjFjPn0AHmUw2MOQDF9bqA8forJwwBMBBXAC%2B-o4QqADCABYsqADmIcD8QiIwUeQBqCwAtiHI4BBsEBal0eZsLABGbgSlAOpJtADk1OgAnhA26LosTJX6GpTllQD81XmOAA51YDAJ5bAWzSCu7nLoLBMVVSA1FBDztPqx1KEMJ%2BZkAQHhMPDoAIzcd451jWyvpdI-CBPnNHugANaGKD-EC0ADuEAAtGxDDBSiCweZ%2BFQoAAZVEwgDEul0AAZyaT0V9MBEMZiHpjMM9XgAmOmMn5uGEAIWCVNBj0hOhh8KRKKM-MZmGxsTxBPQpUJFNJJMlmNp1NImqZ2VeAGZ2ZjOX8FSAAOJWGCoNWMoXQ02i5Gom2YmU6fFGInkknkl0BDUCzCiXIBAD0RGipnIEVweDQuRAESAA

and is different with respect to the behaviour of the  standard HTML select field.

Maybe a new option "isNullable" or similar should activate this behaviour?

Or avoid it if isMulti is false?

**Comment by gabro** [comment]
Hi @mperrando, so you basically want to have a `SelectField` whose value can never be cleared? We can consider adding a `clearable` option to it.

**Reply by mperrando** [reply]
Yes.

Could it be: if `isMulti` is true the user can clear the field deselecting each value; if `isMulti` is false, the field can not be cleared unless the new `clearable` option is set?

That is, the `clearable` is available only when `isMulti` is false?

**Reply by ghubguil** [reply]
@gabro have you considered this issue in your roadmap?
Is there any chance to have a solution in place by the next week (dec 15th)?
Otherwise we need to kinda patch it by ourselves, in some way (@mperrando).

**Reply by gabro** [reply]
We'll look into it and try to get an improvement in by next week!

**Reply by gabro** [reply]
Released in 0.22.2! There is now a `clearable` prop you can set to false

**Reply by ghubguil** [reply]
Thanks @gabro !
